.github/workflows/docker-build-push-dockerhub.yml

```diff
@@ -112,6 +114,43 @@ jobs:
       #   run: |
       #     DOCKER_BUILDKIT=0 docker compose build --build-arg DATABASE_URL=postgresql://${{ env.POSTGRES_USER }}:${{ env.POSTGRES_PASSWORD }}@${{ env.DATABASE_HOST }}/${{ env.POSTGRES_DB }} calcom
 
+      - name: Test runtime
+        run: |
+            docker run --rm --network stack \
+              -p 3000:3000 \
+              -e DATABASE_URL=postgresql://${{ env.POSTGRES_USER }}:${{ env.POSTGRES_PASSWORD }}@database/${{ env.POSTGRES_DB }} \
+              -e NEXTAUTH_SECRET=${{ env.NEXTAUTH_SECRET }} \
+              -e CALENDSO_ENCRYPTION_KEY=${{ env.CALENDSO_ENCRYPTION_KEY }} \
+              docker.io/calcom/cal.com:add-runtime-check &
+              
+            server_pid=$!
+
+
+            echo "Waiting for the server to start..."
+            sleep 120
+
+            echo ${{ env.NEXT_PUBLIC_WEBAPP_URL }}/auth/login
+        
+            for i in {1..60}; do
+              echo "Checking server health ($i/60)..."
+              response=$(curl -o /dev/null -s -w "%{http_code}" ${{ env.NEXT_PUBLIC_WEBAPP_URL }}/auth/login)
+              echo "HTTP Status Code: $response"
+              if [[ "$response" == "200" ]] || [[ "$response" == "307" ]]; then
+                echo "Server is healthy"
+                # Now, shutdown the server
+                kill $server_pid
+                exit 0
+              fi
+              sleep 1
+            done
```

Comment:
Check every second until `true` if we can access `/auth/login`, and we tend to currently redirect it to `/auth/setup` but even that is a good indication of server being functional, so we accept both 200 and 307 to cover this.